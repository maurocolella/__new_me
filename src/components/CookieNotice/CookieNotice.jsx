import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';

import styles from './CookieNotice.scss';

class CookieNotice extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      accepted: localStorage ? localStorage.getItem('cookiesAccepted') : false,
      dismissed: false,
      noticeDelivered: localStorage ? localStorage.getItem('noticeDelivered') : null,
    };
  }

  handleAccept = () => {
    this.setState({
      accepted: true,
      dismissed: true,
    });
  }

  handleDismiss = () => {
    this.setState({
      dismissed: true,
    });
  }

  handleDismissEnd = (event) => {
    if (localStorage && event.target === event.currentTarget) {
      const { accepted } = this.state;
      localStorage.setItem('noticeDelivered', true);
      localStorage.setItem('cookiesAccepted', accepted);
      this.setState({
        noticeDelivered: true,
      });
    }
  }

  render() {
    const { dismissed, noticeDelivered } = this.state;

    return (Boolean(noticeDelivered) === false && (
      <div
        className={`${styles.notice}${dismissed ? ` ${styles['notice--fade']}` : ''}`}
        onTransitionEnd={this.handleDismissEnd}
      >
        <section>
          <button
            onClick={this.handleDismiss}
            className={styles.notice__dismiss}
            type="button"
          >
            <i className="material-icons">close</i>
          </button>
        </section>
        <section className={styles.notice__text}>
          This website uses Google Analytics in order to collect anonymous usage data.
          You can find out more in my
          {' '}
          <Link to="/privacy" className={styles.notice__link}>privacy policy.</Link>
          {' '}
          By using this website, you consent to the use of
          statistical information according to the privacy policy.
        </section>
        <section>
          <button
            onClick={this.handleAccept}
            className={styles.notice__accept}
            type="button"
          >
            Accept
          </button>
        </section>
      </div>
    ));
  }
}

export default CookieNotice;
