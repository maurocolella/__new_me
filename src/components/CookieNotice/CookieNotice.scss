.notice {
  background-color: #001020;
  color: #ddd;
  display: flex;
  font-family: 'Open Sans', sans-serif;
  font-size: 14px;
  justify-content: space-evenly;
  left: 0;
  padding: 24px;
  position: fixed;
  bottom: 0;
  transition: 0.6s opacity ease;
  width: 100%;
  z-index: 10;

  &__link {
    color: #787878;
  }

  &--fade {
    opacity: 0;
  }

  &__text {
    padding: 0 12px;
  }

  &__dismiss {
    background-color: transparent;
    border: 0px;
    color: #787878;
    cursor: pointer;
    float: left;
    outline: 0;
    vertical-align: middle;
  }

  &__accept {
    background-color: transparent;
    border: 1px solid #dfdfdf;
    border-radius: 3px;
    color: #fff;
    cursor: pointer;
    left: 50%;
    margin: 6px;
    outline: 0;
    padding: 6px;
    text-transform: uppercase;
    transition: 0.2s all ease;

    &:active,
    &:hover {
      background-color: #fff;
      color: #333;
    }
  }
}
