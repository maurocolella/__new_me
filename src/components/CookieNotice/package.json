{
  "name": "CookieNotice",
  "version": "0.0.0",
  "private": true,
  "main": "./CookieNotice.jsx"
}
