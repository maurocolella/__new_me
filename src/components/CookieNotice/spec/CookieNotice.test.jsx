import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as MockRouter } from 'react-router-dom';
import CookieNotice from '../CookieNotice';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MockRouter>
        <CookieNotice />
      </MockRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
