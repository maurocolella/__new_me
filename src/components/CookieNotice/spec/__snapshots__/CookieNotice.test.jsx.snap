// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="notice"
  onTransitionEnd={[Function]}
>
  <section>
    <button
      className="notice__dismiss"
      onClick={[Function]}
      type="button"
    >
      <i
        className="material-icons"
      >
        close
      </i>
    </button>
  </section>
  <section
    className="notice__text"
  >
    This website uses Google Analytics in order to collect anonymous usage data. You can find out more in my
     
    <a
      className="notice__link"
      href="/privacy"
      onClick={[Function]}
    >
      privacy policy.
    </a>
     
    By using this website, you consent to the use of statistical information according to the privacy policy.
  </section>
  <section>
    <button
      className="notice__accept"
      onClick={[Function]}
      type="button"
    >
      Accept
    </button>
  </section>
</div>
`;
